google/ggrc-core
# GGRC-5532 Remove double "destroy" call when Control's edit modal is saving

# Dependencies

This PR is `on hold` until the dependencies are merged:

- [x] #8114


# Issue description

"Uncaught TypeError: Cannot read property 'find' of null" occurs if click propose changes button and then click 'esc'

# Steps to test the changes

Steps to reproduce:
1. Open any control page
2. Click Propose changes button
3. While Edit control modal window is loading press 'esc' on keyboard
*Actual Result:* "Uncaught TypeError: Cannot read property 'find' of null" occurs if click propose changes button and then click 'esc' 
*Expected Result:* no errors should be displayed if click propose changes button and then click 'esc' 

Notes:
I haven't reproduced this issue through provided steps.
But after some investigation, I've understood this issue is appeared, when we do the following steps:
1. Open Edit modal for any object.
2. Click Hide optional fields.
3. Save and Close.
4. Open Edit modal for previous object again.
4.1 During the loading of edit modal, quickly click esc key.

*Actual Result:* "Uncaught TypeError: Cannot read property 'find' of null" occurs if click propose changes button and then click 'esc' 
*Expected Result:* no errors should be displayed if click propose changes button and then click 'esc' 

# Solution description

This issue was appeared due to quick clicking using esc button. Info pane has quick_form_controller and calls 'destroy' method, when we close info pane. quick_form_controller is based on modals_controller, but doesn't call this._super() in constructor (modals_controller::init method). It leads to situation, when some data is initialized into base constructor (modals_controller::init method) and used in modals_controller::destroy method. quick_form_controller doesn't have own implementation for `destroy` method. Because of that, it uses the method from parent (modals_controller::destroy). As far as we can see this method will use uninitialized data.

Was removed quick_form_controller.
Some functionality was moved from quick_form_controller to approval link component.
Was created an util, which provides ability to stop async chain.

# Sanity checklist




# PR Review checklist
